lib/tpm2_alg_util.c

```diff
@@ -880,6 +880,16 @@ bool tpm2_alg_util_public_init(char *alg_details, char *name_halg, char *attrs,
         }
     }
 
+    /* load a policy from a path if present */
+    if (unique_file) {
+        UINT16 unique_size = sizeof(public->publicArea.unique);
+        bool res = files_load_bytes_from_path(unique_file,
+                    (UINT8*)&public->publicArea.unique, &unique_size);
+        if (!res) {
+            return false;
+        }
```

Comment:
this can return true on a smaller than `sizeof(public->publicArea.unique);` read, is that ok?

Reply:
Yes it is. It is a union field. Added inline comment.